pkg/kube/resources_test.go

```diff
@@ -99,7 +96,19 @@ func TestGetResourceFromAPI(t *testing.T) {
 	assert.IsType(t, time.Now(), resources.CreationTime, "Creation time should be set")
 
 	assert.Equal(t, 0, len(resources.Nodes), "Should not have any nodes")
-	assert.Equal(t, 1, len(resources.Controllers), "Should have 1 controller")
-
-	assert.Equal(t, "", resources.Controllers[0].ObjectMeta.GetName())
+	assert.Equal(t, 5, len(resources.Controllers), "Should have 5 controllers")
+
+	expectedNames := map[string]bool{
+		"deploy":      false,
+		"job":         false,
+		"cronjob":     false,
+		"statefulset": false,
+		"daemonset":   false,
+	}
+	for _, ctrl := range resources.Controllers {
+		expectedNames[ctrl.ObjectMeta.GetName()] = true
```

Comment:
I like this way of handling. I wonder if the assert library we use has something built in already or if we could contribute to it

Reply:
I think there's something like it, but it expects the same object type